keps/sig-node/20191025-image-pull-progress.md

```diff
@@ -0,0 +1,402 @@
+thrown events from the kubelet during an image pull. This should be limited in a
+fashion that the API surface is not strongly impacted.
+
+### Risks and Mitigations
+
+The interface changes to `kubectl` should increase the usability to the users,
+whereas it opens the doors for future correlations between events and
+`kubectl get` requests. There is a risk that the current design proposal does
+not fulfill that need.
+
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+
+- **How can this feature be enabled / disabled in a live cluster?**
+
+  Via the kubelet dynamic configuration feature, where we will add a dedicated
+  option for it.
+
+- **Can the feature be disabled once it has been enabled (i.e., can we roll
+  back the enablement)?**
+
+  Yes, but not during a image pull (already established gRPC stream).
+
+- **Will enabling / disabling the feature require downtime for the control
+  plane?**
+
+  No
+
+- **Will enabling / disabling the feature require downtime or reprovisioning
+  of a node?**
+
+  No
+
+- **What happens if a cluster with this feature enabled is rolled back? What
+  happens if it is subsequently upgraded again?**
+
+  Nothing, since the configuration option (enabled or disabled) is only a
+  toggle and relies on no other state.
+
+- **Are there tests for this?**
+
+  Yes, end to end tests for the kubelet.
+
+### Scalability
+
+- **Will enabling / using the feature result in any new API calls?
+  Describe them with their impact keeping in mind the supported limits
+  (e.g. 5000 nodes per cluster, 100 pods/s churn) focusing mostly on:**
+
+  - **components listing and/or watching resources they didn't before**
+  - **API calls that may be triggered by changes of some Kubernetes
+    resources (e.g. update object X based on changes of object Y)**
+  - **periodic API calls to reconcile state (e.g. periodic fetching state,
+    heartbeats, leader election, etc.)**
+
+  Yes, we throw new events during the image pull process. The amount can be
+  configured on the kubelet as well. The data for the events are coming from
+  the corresponding container runtime, which streams the result to the
+  kubelet in the same amount. There is no further state involved than the
+  events one.
+
+  **Example:** We assume an average container image size of 100 MiB and an
+  available bandwidth of 100 MBit/s. This means that the overall image
+  transfer from the remote registry would take around 8 seconds.
+
+  Now we assume a workload throughput of 100 new Pods/s with uniquely used
+  images, so we have a window of round about 800 pods downloading a container
+  image at the same time.
+
+  In any configuration case we would receive a starting and ending event for
+  the image pull, which would result in at least 200 QPS to the API Server.
+
+  If we configure the feature to an event period of 1 second, then it would
+  result in 800 QPS to the API Server.
```

Comment:
I wouldn't give single number - instead I would say that depending on the interval it will be between 0 and 800 additinal QPS in this scenario.

Reply:
Yes I agree, fixed.